Source/MultipartFormData.swift

```diff
@@ -189,43 +189,22 @@ open class MultipartFormData {
     ///
     /// The body part data will be encoded using the following format:
     ///
-    /// - `Content-Disposition: form-data; name=#{name}; filename=#{generated filename}` (HTTP Header)
-    /// - `Content-Type: #{generated mimeType}` (HTTP Header)
+    /// - `Content-Disposition: form-data; name=#{name}; filename=#{filename}` (HTTP Header)
+    /// - `Content-Type: #{mimeType}` (HTTP Header)
     /// - Encoded file data
     /// - Multipart form boundary
     ///
-    /// The filename in the `Content-Disposition` HTTP header is generated from the last path component of the
-    /// `fileURL`. The `Content-Type` HTTP header MIME type is generated by mapping the `fileURL` extension to the
-    /// system associated MIME type.
-    ///
-    /// - parameter fileURL: The URL of the file whose content will be encoded into the multipart form data.
-    /// - parameter name:    The name to associate with the file content in the `Content-Disposition` HTTP header.
-    public func append(_ fileURL: URL, withName name: String) {
-        let fileName = fileURL.lastPathComponent
-        let pathExtension = fileURL.pathExtension
-
-        if !fileName.isEmpty && !pathExtension.isEmpty {
-            let mime = mimeType(forPathExtension: pathExtension)
-            append(fileURL, withName: name, fileName: fileName, mimeType: mime)
-        } else {
-            setBodyPartError(withReason: .bodyPartFilenameInvalid(in: fileURL))
-        }
-    }
-
-    /// Creates a body part from the file and appends it to the multipart form data object.
-    ///
-    /// The body part data will be encoded using the following format:
-    ///
-    /// - Content-Disposition: form-data; name=#{name}; filename=#{filename} (HTTP Header)
-    /// - Content-Type: #{mimeType} (HTTP Header)
-    /// - Encoded file data
-    /// - Multipart form boundary
+    /// The filename in the `Content-Disposition` HTTP header, if needed, is generated from the last path component of the
+    /// `fileURL`. The `Content-Type` HTTP header MIME type, if needed, is generated by mapping the `fileURL` extension to the
+    /// system associated MIME types.
     ///
     /// - parameter fileURL:  The URL of the file whose content will be encoded into the multipart form data.
     /// - parameter name:     The name to associate with the file content in the `Content-Disposition` HTTP header.
     /// - parameter fileName: The filename to associate with the file content in the `Content-Disposition` HTTP header.
     /// - parameter mimeType: The MIME type to associate with the file content in the `Content-Type` HTTP header.
-    public func append(_ fileURL: URL, withName name: String, fileName: String, mimeType: String) {
+    public func append(_ fileURL: URL, withName name: String, fileName: String? = nil, mimeType: String? = nil) {
+        let fileName = fileName ?? fileURL.lastPathComponent
```

Comment:
...here `fileName` gets checked, if `nil` we use `fileURL.lastPathComponent` as before.